### PR TITLE
More debugging for issue where user cannot send all messages

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

We have a couple users who are unable to send all their messages.  Actually the UI reports that their messages are sent, but we can see in the logs that they are are not, and that an 'Invalid Phone Number' exception is being thrown.

We don't know yet know the cause, but one possibility is that the csv file that we autogenerate when they do a one-off send is being corrupted.   However, right now we don't have a way to see that csv file.

Add a command so that that we can download csv files by name from production.

## Security Considerations

User can only run the command to download a csv file on production if they:

1. Know where to find the file (which means they already have access to notify-admin-production logs
2. Are able to run the command 'cf run-task notify-api "flask command download-civ-file-by-name -f <file>", which means they have already authenticated through cloud.gov

### CODE REVIEW

File: docs/all.md [hunk 1]
```diff
@@ -1224,3 +1227,25 @@ Data Retention Policy
...
+
+Ask the user to provide the csv file name.  Either the csv file they uploaded, or the one that is autogenerated when they do a one-off send and is visible in the UI
+
+Starting with the admin logs, search for this file name.  When you find it, the log line should have the file name linked to the job_id and the csv file location.  Save both of these.
+
+In the api logs, search by job_id.  Either you will see evidence of the job failing and retrying over and over (in which case search for a stack trace using timestamp), or you will ultimately get to a log line that links the job_id to a message_id.  In this case, now search by message_id.  You should be able to find the actual result from AWS, either success or failure, with hopefully some helpful info.
+
+## Viewing the csv file
+
+If you need to view th questionable csv file, run the following command:
```
Comment (ccostino) [inline docs/all.md:1246]: Small typo:

```suggestion
If you need to view the questionable csv file, run the following command:
```

File: app/commands.py [hunk 2]
```diff
@@ -589,6 +589,17 @@ def process_row_from_job(job_id, job_row_number):
...
 
 
+@notify_command(name="download-csv-file-by-name")
+@click.option("-f", "--csv_filename", required=True, help="csv file name")
+def download_csv_file_by_name(csv_filename):
+
+    bucket_name = getenv("CSV_BUCKET_NAME")
+    access_key = getenv("CSV_AWS_ACCESS_KEY_ID")
+    secret = getenv("CSV_AWS_SECRET_ACCESS_KEY")
+    region = getenv("CSV_AWS_REGION")
```
Comment (ccostino) [inline app/commands.py:599]: Are these environment variables the ones that come from our own `.env` files?  If so, they are not available in the cloud.gov environment so this won't work in a `cf run-task` context.  These should be coming from the application configuration instead, which takes care of retrieving the correct environment variables regardless of the environment.

You can see how these are referenced in the [`app/aws/s3.py`](https://github.com/GSA/notifications-api/blob/main/app/aws/s3.py#L61-L68) file, specifically the `get_job_location` method.

In fact, should we be using something like `get_s3_object` from that `aws/s3.py` file in place here?

**Review by ccostino — CHANGES_REQUESTED** [review]
Thanks, @terrazoon, hopefully this will help in tracking down what's going on!

There was something I noticed with the command that I think we need to adjust given the use of environment variables - the values need to come from the app config instead.

This PR also just needs a quick update with `main` again given the `poetry.lock` file still being present in here and that was updated to hopefully keep it from changing constantly for folks.

File: app/commands.py (hunk 2 above)
Comment (terrazoon) [reply app/commands.py:599]: we are calling s3.get_s3_file() which in turn calls s3.get_s3_object()

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon!

File: app/commands.py (hunk 2 above)
Comment (ccostino) [reply app/commands.py]: D'oh, you're right!  My mistake there, I mis-read the line!

**Pull request merged**